packages/blade/src/components/Skeleton/_decisions/decisions.md

```diff
@@ -0,0 +1,320 @@
+            />
+          </CardHeader>
+          <CardBody>
+            <Text>{data.content}</Text>
+          </CardBody>
+        </Card>
+      )}
+    </>
+  );
+};
+```
+
+If only body of the card needs to be lodable consumers can put the skeleton on the CardBody:
+
+```jsx
+const App = () => {
+  const { data, isLoading } = useQuery();
+
+  return (
+    <Card>
+      <CardHeader>
+        <CardHeaderLeading title="Payment options" subtitle="Secured by Razorpay" />
+        <CardHeaderTrailing visual={<CardHeaderBadge variant="neutral">PENDING</CardHeaderBadge>} />
+      </CardHeader>
+      <CardBody>
+        {isLoading ? <Skeleton width="100%" height="100px" /> : <Text>{data.content}</Text>}
+      </CardBody>
+    </Card>
+  );
+};
+
+```
+
+**Q. A question that might arise is "Why not just give isLoading prop for Card?"**
+
+Like: 
+
+```jsx
+<Card isLoading={isLoading}>
+  <CardHeader>
+    <CardHeaderLeading title={data.title} subtitle={data.subtitle} />
+    <CardHeaderTrailing visual={<CardHeaderBadge variant="neutral">{data.badge}</CardHeaderBadge>} />
+  </CardHeader>
+  <CardBody>
+    <Text>{data.content}</Text>
+  </CardBody>
+</Card>
+```
+
+We discussed this, while it might look simple this approach also has few downsides: 
+
+- if we provide `isLoading` prop in Card it might cause a lot of jumps and shifts on the page because we won't know until API responds if the CardHeader has title/prefix or subtitle set or not and it could be removed after the API is done loading.
+- Providing `isLoading` prop in Card or for all Blade components (where applicable) might not be the most flexible approach.
+- Providing `isLoading` prop in Card consumers will anyways have to opt out and use custom Skeleton templates for the CardBody content so there's not much benefit to providing it out of the box.
+
+
+## Motion
+
+Check the motion [here](https://www.figma.com/proto/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=16502-258345&scaling=min-zoom&page-id=16498%3A256331):
+
+https://github.com/razorpay/blade/assets/35374649/8298efde-f977-4aa3-bc2d-92fb50a8c830
+
+
+## Accessibility
+
+- All Skeleton nodes will have aria-hidden to hide them from screen readers
+- All Skeleton nodes will have `cursor: progress`
+- Consumers will have to add aria-busy in the respective container which is being rendered
+  - We cannot automatically added this because we will end up adding multiple aria-busy atts to all the <Skeleton \/> nodes.
+- Or: consumers can also use the <VisuallyHidden \/> element to add hidden `loading...` text in DOM.
+
+> Reference: https://adrianroselli.com/2020/11/more-accessible-skeletons.html
+
+## Open Questions
+
+- How will Skeleton loader work with existing components like ModalHeader/Footer etc?
+  - Ans: we will let the consumer handle the skeleton loading for Card components
+
+
+<details>
+
+  <summary>
+    Should we infer dimensions or provide predefined variants?
```

Comment:
Wondering if we should provide couple of predefined variants like circle, rectangle which would automatically add required properties such as border radius for circle would be 50%, for rectangle border radius will be say 4px. Otherwise there might be lot of repetition required where people have to pass border radius 4px to any rectangular skeleton they're using (plus if in future we want to tweak that).

Another common usecase that comes to mind is where people want to show skeleton loaders for a piece of text they're rendering. Since we don't have a predefined variant for it, consumers will have to handle spacing, number of lines of skeleton loader, etc. Slight inconvenience.

Reply:
> Another common usecase that comes to mind is where people want to show skeleton loaders for a piece of text

Yeah we actually had a long discussion on this please check the "Should we infer dimensions" open question where we explored this and why we decided not to do it as phase-1

Reply:
Is this planned as future enhancement? Can be added in the doc.